nisystemlink/clients/dataframe/utilities/_pandas_utils.py

```diff
@@ -0,0 +1,157 @@
+from typing import List, Optional, Union
+
+import pandas as pd
+from nisystemlink.clients.dataframe import DataFrameClient
+from nisystemlink.clients.dataframe.models import Column, ColumnType, DataType
+
+from ._pandas_exception import InvalidColumnTypeError, InvalidIndexError
+
+UNSUPPORTED_PANDAS_INT_TYPES = ["int8", "int16"]
+"""List of unsupported pandas integer types for conversion to `DataType`."""
+
+UNSUPPORTED_PANDAS_FLOAT_TYPES = ["float16"]
+"""List of unsupported pandas float types for conversion to `DataType`."""
+
+SUPPORTED_INDEX_DATA_TYPE = [DataType.Int32, DataType.Int64, DataType.Timestamp]
+"""List of supported index data types for table creation.
+
+Only these `DataType` values are allowed for the index column of the table.
+"""
+SUPPORTED_PANDAS_DATATYPE_MAPPING = {
+    "bool": DataType.Bool,
+    "int32": DataType.Int32,
+    "int64": DataType.Int64,
+    "float32": DataType.Float32,
+    "float64": DataType.Float64,
+    "object": DataType.String,
+    "datetime64[ns]": DataType.Timestamp,
+}
+"""Mapping of pandas data types to `DataType`.
+
+This dictionary maps commonly used pandas data types to the corresponding `DataType` used in table creation.
+"""
+
+
+def _pandas_dtype_to_data_type(dtype: str) -> Optional[DataType]:
+    """Convert pandas data type to `DataType`.
+
+    Args:
+        dtype (str): Pandas data type.
+
+    Returns:
+        Optional[DataType]: `DataType`or `None` if match not found.
+    """
+    if dtype in SUPPORTED_PANDAS_DATATYPE_MAPPING:
+        return SUPPORTED_PANDAS_DATATYPE_MAPPING[dtype]
+    return None
+
+
+def _type_cast_column_datatype(
+    data: Union[pd.Index, pd.Series]
+) -> Union[pd.Index, pd.Series]:
+    """Process data to convert to supported type if necessary.
+
+    Args:
+        data (Union[pd.Index, pd.Series]): Data to be processed.
+
+    Returns:
+        Union[pd.Index, pd.Series]: Processed data.
+    """
+    pd_dtype = data.dtype
+    if pd.api.types.is_unsigned_integer_dtype(data):
+        data = pd.to_numeric(data, downcast="integer")
+        pd_dtype = data.dtype
+
+    if pd_dtype in UNSUPPORTED_PANDAS_INT_TYPES:
```

Comment:
It would be good to make the `UNSUPPORTED_PANDAS_INT_TYPES` and `UNSUPPORTED_PANDAS_FLOAT_TYPES` into dict that maps to the bumped up type to cast into